### PR TITLE
Make Mesos CI builds interruptible

### PR DESCRIPTION
**Pull request by karlkfi**
interruptible with sigterm or sigint

Unfortunately, this makes the in-line dots not work for indicating progress, because the stdout log tail only does full lines at a time.

### CODE REVIEW

File: contrib/mesos/ci/run-with-cluster.sh [hunk 1]
```diff
@@ -56,30 +56,49 @@ echo "${DOCKER_BIN_PATH}"
...
+    -e "USER=root" \
+    -e "E2E_REPORT_DIR=${MESOS_DOCKER_WORK_DIR}/reports" \
+    mesosphere/kubernetes-mesos-test \
+    -ceux "\
+      make clean all && \
+      trap 'timeout 5m ./cluster/kube-down.sh' EXIT && \
+      ./cluster/kube-up.sh && \
+      trap 'test \$? != 0 && export MESOS_DOCKER_DUMP_LOGS=true; timeout 5m ./cluster/kube-down.sh' EXIT && \
+      ${RUN_CMD}
+    "
```
Comment (sttts) [inline contrib/mesos/ci/run-with-cluster.sh]: Why not make this interactive with `-i`. Then docker does everything about signal handling for us.

**Comment by sttts** [comment]
Compare comment. Why is `-i` not enough? Why do we need this amount of additional script code?

**Comment by sttts** [comment]
@karlkfi compare https://github.com/kubernetes/kubernetes/pull/17842

**Comment by karlkfi** [comment]
Replaced by #17842

**Pull request closed without merging**